src/workerd/io/io-context.c++

```diff
@@ -949,6 +949,14 @@ void IoContext::taskFailed(kj::Exception&& exception) {
       waitUntilStatusValue = *status;
     } else {
       waitUntilStatusValue = EventOutcome::EXCEPTION;
+      if (auto desc = exception.getDescription();
+          !jsg::isTunneledException(desc) && !jsg::isDoNotLogException(desc)) {
+        if (isInterestingException(exception)) { // Not sure on this one
+          LOG_EXCEPTION("IoContextErrorHandler"_kj, exception);
```

Comment:
Can you please update the PR description to explain the motivation for this change? What kinds of errors are failing to get logged, exactly, that are caught by this? Why only log in this particular branch as opposed to higher up in this function?